tools/fapi/tss2_template.c

```diff
@@ -250,6 +250,75 @@ TSS2_RC auth_callback(
     return TSS2_RC_SUCCESS;
 }
 
+TSS2_RC sign_callback(
+    char        const *description,
+    char        const *objectPath,
+    char        const *publicKey,
+    char        const *publicKeyHint,
+    uint32_t    hashAlg,
+    uint8_t     const *dataToSign,
+    size_t      dataToSignSize,
+    uint8_t     const **signature,
+    size_t      *signatureSize,
+    __attribute__((unused)) void *userData)
+{
+    /* The public key that will be used by the TPM to verify the signature in PEM encoding */
+    (void) publicKey; // 
+    (void) publicKeyHint;
+    (void) hashAlg;
+    (void) dataToSign;
+    (void) dataToSignSize;
+
+    int rc;
+    char *default_name = "Data2Sign.file";
+    char buf[1024];
+
+    if (userData) {
+        printf("%s:", (const char *) userData);
+    } else {
+        printf ("Authorize %s with \"%s\"\n", description, objectPath);
+    }
+    printf("Filename for nonce output: ");
+    if (buf == fgets(buf, 1024, stdin)) {
+        char path[256];
+        rc = sscanf(buf, "%s", path);
```

Comment:
We limit the input to 1024 but then the path to 256? Why not just have them the same size, am I missing something?

Reply:
Fixed in current version.